lib/sles4sap/aws_cli.pm

```diff
@@ -0,0 +1,494 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Library wrapper around some aws cli commands.
+
+package sles4sap::aws_cli;
+use strict;
+use warnings FATAL => 'all';
+use testapi;
+use Carp qw(croak);
+use Exporter qw(import);
+use Mojo::JSON qw(decode_json);
+use Regexp::Common qw(net);
+use NetAddr::IP;
+use utils qw(write_sut_file);
+
+
+=head1 SYNOPSIS
+
+Library to compose and run AWS cli commands
+=cut
+
+our @EXPORT = qw(
+  aws_import_key_pair
+  aws_create_vpc
+  aws_get_vpc_id
+  aws_create_security_group
+  aws_get_security_group_id
+  aws_create_subnet
+  aws_get_subnet_id
+  aws_create_internet_gateway
+  aws_get_internet_gateway_id
+  aws_attach_internet_gateway
+  aws_create_route_table
+  aws_associate_route_table
+  aws_create_route
+  aws_authorize_security_group_ingress
+  aws_create_vm
+  aws_get_vm_id
+  aws_wait_instance_status_ok
+  aws_get_ip_address
+);
+
+
+=head2 aws_import_key_pair
+
+    aws_import_key_pair($ssh_key, $pub_key_path);
+
+Import an SSH public key pair into AWS EC2 for instance authentication
+
+=over
+
+=item B<$ssh_key> - name to assign to the imported key pair in AWS
+
+=item B<$pub_key_path> - filesystem path to the public key file
+
+=back
+=cut
+
+sub aws_import_key_pair {
+    my ($ssh_key, $pub_key_path) = @_;
+    assert_script_run("aws ec2 import-key-pair --key-name '$ssh_key' --public-key-material fileb://$pub_key_path");
+}
+
+=head2 aws_create_vpc
+
+    my $vpc_id = aws_create_vpc($region, $cidr, $job_id);
+
+Create a new AWS VPC with a specified CIDR block and tag it with the OpenQA job ID
+
+=over
+
+=item B<$region> - AWS region where to create the VPC
+
+=item B<$cidr> - CIDR block for the VPC (e.g., '10.0.0.0/16')
+
+=item B<$job_id> - OpenQA job identifier for tagging
+
+=back
+
+Returns the VPC ID
+=cut
+
+sub aws_create_vpc {
+    my ($region, $cidr, $job_id) = @_;
+    my $vpc_id = script_output("aws ec2 create-vpc --region $region --cidr-block $cidr --query 'Vpc.VpcId' --output text");
```

Comment:
Think about adding some sort of validation on the returned `$vpc_id` value before using it to compose the next command.